rotkehlchen/db/updates.py

```diff
@@ -154,6 +163,70 @@ def update_rpc_nodes(self, to_version: int) -> None:
             new_default_nodes=new_default_nodes,
         )
 
+    def update_global_addressbook(self, from_version: int, to_version: int) -> None:
+        """Downloads and applies global addressbook updates from from_version to to_version"""
+        db_addressbook = DBAddressbook(self.user_db)
+        for version in range(from_version + 1, to_version + 1):
+            log.debug(f'Applying global address book update version {version}')
+            success, new_raw_entries = self._get_file_content(
+                file_url=GLOBAL_ADDRESSBOOK_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.GLOBAL_ADDRESSBOOK,
+            )
+            if success is False:
+                log.error(f'Failed to download globaldb addressbook for version {version}')
+                continue
+
+            entries_to_add = []
+            for raw_entry in new_raw_entries:
+                try:
+                    address = deserialize_evm_address(raw_entry['address'])
+                    blockchain = SupportedBlockchain.deserialize(raw_entry['blockchain']) if raw_entry['blockchain'] else None  # noqa: E501
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'Could not deserialize address {raw_entry["address"]} or blockchain '
+                        f'{raw_entry["blockchain"]} that was seen in a global addressbook update. '
+                        f'Please report it to the rotki team. {str(e)}',
+                    )
+                    continue
+
+                entry = AddressbookEntry(
+                    address=address,
+                    name=raw_entry['new_name'],
+                    blockchain=blockchain,
+                )
+                existing_name = db_addressbook.get_addressbook_entry_name(
+                    book_type=AddressbookType.GLOBAL,
+                    chain_address=OptionalChainAddress(
+                        address=address,
+                        blockchain=blockchain,
+                    ),
+                )
+                if existing_name is None:
+                    entries_to_add.append(entry)
+                elif raw_entry['old_name'] is not None and existing_name == raw_entry['old_name']:
```

Comment:
If for some reason we miss it you were gonna get a key error here

```suggestion
                elif raw_entry.get('old_name') is not None and existing_name == raw_entry['old_name']:
```